cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -209,40 +223,14 @@ public long getLastSequence() {
         Preconditions.checkState(this.isOpen(), "Database is closed");
 
         try {
-
-            return queue.submit(new SQLCallable<Long>() {
-                @Override
-                public Long call(SQLDatabase db) throws Exception {
-                    String sql = "SELECT MAX(sequence) FROM revs";
-                    Cursor cursor = null;
-                    long result = 0;
-                    try {
-                        cursor = db.rawQuery(sql, null);
-                        if (cursor.moveToFirst()) {
-                            if (cursor.columnType(0) == Cursor.FIELD_TYPE_INTEGER) {
-                                result = cursor.getLong(0);
-                            } else if (cursor.columnType(0) == Cursor.FIELD_TYPE_NULL) {
-                                result = SEQUENCE_NUMBER_START;
-                            } else {
-                                throw new IllegalStateException("SQLite return an unexpected value.");
-                            }
-                        }
-                    } catch (SQLException e) {
-                        logger.log(Level.SEVERE, "Error getting last sequence", e);
-                        throw new DatastoreException(e);
-                    } finally {
-                        DatabaseUtils.closeCursorQuietly(cursor);
-                    }
-                    return result;
-                }
-            }).get();
+            return queue.submit(new GetLastSequenceCallable()).get();
         } catch (InterruptedException e) {
-            logger.log(Level.SEVERE,"Failed to get last Sequence",e);
+            logger.log(Level.SEVERE, "Failed to get last Sequence", e);
             throw new RuntimeException(e);
         } catch (ExecutionException e) {
             logger.log(Level.SEVERE, "Failed to get last Sequence", e);
-            if(e.getCause()!=null){
-                if(e.getCause() instanceof IllegalStateException){
+            if (e.getCause() != null) {
+                if (e.getCause() instanceof IllegalStateException) {
```

Comment:
Looking at how many times we write the pattern:
```java
try {
  return queue.submit(callable).get();
} catch (InterruptedException e) {
            logger.log(Level.SEVERE, "Identical message", e);
             throw new RuntimeException(e);
} catch (ExecutionException e) {
             logger.log(Level.SEVERE, "Identical message", e);
}
```
I'd be sorely tempted to wrap the `Future` returned from the `SQLDatabaseQueue.submit*` methods in another type with a modified get that does this try/catch/log for us.